### DISCUSSION TITLE
How to burn multiple roms into nic?

### DISCUSSION THREAD

**Discussion by xuchao1234** [ipxe/ipxe#894] [Q&A] [answered]
I made several images, for example, bin/808610de. rom, bin-x86_ 64-efi/808610de. efirom, bin-arm64-efi/808610de.efirom, so that my network card can be used in i386-pcbios, x86_ 64-efi and arm64-efi. Is there any way to integrate these three images into one so that I can burn it into the pci expansion rom?

**Comment by mcb30** [comment, marked as answer]
Yes: the script in `./util/catrom.pl` can do that for you.  For example:

```
./util/catrom.pl bin/808610de.rom bin-x86_ 64-efi/808610de.efirom bin-arm64-efi/808610de.efirom > combined.rom
```

**Reply by mcb30** [reply]
You may also want to use `bin/808610de.mrom` instead of `bin/808610de.rom`: that will conserve option ROM space in the BIOS build environment.  (Not all cards can support the `.mrom` format, but that one can.)

**Reply by xuchao1234** [reply]
Your suggestion has solved my problem. Thank you very much!

**Reply by stappersg** [reply]
> Your suggestion has solved my problem.

Does who can mark this discussion as _answered_, please do.
